velox/expression/ExprCompiler.cpp

```diff
@@ -329,6 +329,9 @@ ExprPtr tryFoldIfConstant(const ExprPtr& expr, Scope* scope) {
     // instance, if other arguments are all null in a function with default null
     // behavior), the query won't fail.
     catch (const std::exception&) {
+      // Clear left-behind values. Simplified path will error out if
+      // not cleared.
+      expr->inputValues().clear();
```

Comment:
@oerling I assume you are running into VELOX_CHECK(result == nullptr) in ConstantExpr::evalSpecialFormSimplified

```
void ConstantExpr::evalSpecialFormSimplified(
    const SelectivityVector& rows,
    EvalCtx& context,
    VectorPtr& result) {
  ...
  // Simplified path should never ask us to write to a vector that was already
  // pre-allocated.
  VELOX_CHECK(result == nullptr);
```

The comment says that "Simplified path should never ask us to write to a vector that was already pre-allocated." That makes sense, but then the fix should probably be in Expr::evalSimplifiedImpl where we use inputValues_ for the 'result'.

Perhaps, we could inputValues_.clear() before inputValues_.resize(inputs_.size()) there.

```
void Expr::evalSimplifiedImpl(
    const SelectivityVector& rows,
    EvalCtx& context,
    VectorPtr& result) {
  // Handle special form expressions.
  if (isSpecialForm()) {
    evalSpecialFormSimplified(rows, context, result);
    return;
  }

  SelectivityVector remainingRows = rows;
  inputValues_.clear();
  inputValues_.resize(inputs_.size());
  const bool defaultNulls = vectorFunction_->isDefaultNullBehavior();

  for (int32_t i = 0; i < inputs_.size(); ++i) {
    auto& inputValue = inputValues_[i];
    inputs_[i]->evalSimplified(remainingRows, context, inputValue);
```